<launch>
    <!-- 
        Supported robot types:         
        * robot_type: weston/agilex/vbot/unitree_a1
    -->
    <arg name="port_name" default="can0" />
    <arg name="robot_type" default="weston" />
    <arg name="base_frame" default="base_link" />
    <arg name="odom_frame" default="odom" />
    <arg name="odom_topic_name" default="odom" />

    <node name="mobile_base_node" pkg="wrp_ros" type="mobile_base_node" output="screen">
        <param name="port_name" type="string" value="$(arg port_name)" />
        <param name="robot_type" type="string" value="$(arg robot_type)" />
        <param name="base_frame" type="string" value="$(arg base_frame)" />
        <param name="odom_frame" type="string" value="$(arg odom_frame)" />
        <param name="odom_topic_name" type="string" value="$(arg odom_topic_name)" />
    </node>
</launch>
